code/game/objects/items/misc.dm

```diff
@@ -22,3 +23,79 @@
 	throw_speed = 1
 	throw_range = 20
 	flags = CONDUCT
+
+/obj/item/petcollar
+	name = "pet collar"
+	desc = "The latest fashion accessory for your favorite pets!"
+	icon_state = "petcollar"
+	item_color = "petcollar"
+	var/tagname = null
+	var/obj/item/card/id/access_id
+
+/obj/item/petcollar/Destroy()
+	QDEL_NULL(access_id)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/petcollar/attack_self(mob/user as mob)
+	var/option = "Change Name"
+	if(access_id)
+		option = input(user, "What do you want to do?", "[src]", option) as null|anything in list("Change Name", "Remove ID")
```

Comment:
While you're here. Can you add a sanity check after the input? It is possible that everything changed between the start of the users input and the end. So the collar can be moved, destroyed etc. Probably those two are the most important

Reply:
How's this ( ͡° ͜ʖ ͡°)